addons/account/models/account_payment.py

```diff
@@ -667,6 +669,80 @@ def _compute_payment_receipt_title(self):
         """ To override in order to change the title displayed on the payment receipt report """
         self.payment_receipt_title = _('Payment Receipt')
 
+    @api.depends('partner_id', 'amount', 'date', 'payment_type')
+    def _compute_duplicated_payment_ids(self):
+        """ Retrieve ids of payments with same partner_id, amount and date as the current payment """
+        payment_to_duplicate_payment = self._fetch_duplicate_reference()
+        for payment in self:
+            # Uses payment._origin.id to handle records in edition/existing records and 0 for new records
+            payment.duplicated_payment_ids = payment_to_duplicate_payment.get(payment._origin, self.env['account.payment'])
+
+    def _fetch_duplicate_reference(self, matching_states=('draft', 'posted')):
+        # Does not perform unnecessary check if partner_id or amount are not set, nor if payment is posted
```

Comment:
should be checking **only** on the account.move.line on the same account that the payment will create, because some transactions may be there, not related to a payment (if we don't register the payment, but directly the bank transaction, for example)

Reply:
Not sure if I understand, so to confirm: as of now, if there is a bank transaction only, a new payment does not identify the transaction as a duplicate. But you mean that it should identify a duplicate if there is a match in partner/amount/date between payment and bank transaction?